test/specs/SwitchToFrameSpec.ts

```diff
@@ -0,0 +1,35 @@
+import { assert, expect } from 'chai';
+import { BrowserUtils } from '../../src/commons/BrowserUtils';
+import { describeCommon } from '../TestHelper';
+
+const FRAME_2_HEADER_1: string = "//*[@id='frame2Header1']";
+const TEXT_OUTSIDE_OF_FRAMES: string = "//*[@id='textOutsideOfFrames']";
+const FRAME_ONE: string = "//*[@id='iframe1']";
+const FRAME_TWO: string = "//*[@id='iframe2']";
+const FRAME_DONT_EXISTS: string = "//*[@id='iframeDontExists']";
+/**
+ * wdio-allure-ts Switch to iFrame actions
+ */
+describeCommon('switchToIframe', () => {
+  it('Header Text At Frame 2', () => {
+    BrowserUtils.notVisible(FRAME_2_HEADER_1);
+    BrowserUtils.switchToFrame(FRAME_TWO);
+    assert.equal($(FRAME_2_HEADER_1).getText(), 'Frame 2 Heading 1');
+  });
+
+  it('Header text at frame 2 not available after switch to frame 1', () => {
+    BrowserUtils.notVisible(FRAME_2_HEADER_1);
+    BrowserUtils.switchToFrame(FRAME_ONE);
+    expect(() => $(FRAME_2_HEADER_1).getText()).to.throw(Error);
+  });
+
+  it('Header Text OutSide Of Frames Fail After Switch To', () => {
+    BrowserUtils.waitForDisplayed(TEXT_OUTSIDE_OF_FRAMES);
+    BrowserUtils.switchToFrame(FRAME_ONE);
+    expect(() => $(TEXT_OUTSIDE_OF_FRAMES).getText()).to.throw(Error);
+  });
+
+  it('Switch To Frame Wrong Xpath', () => {
+    expect(() => BrowserUtils.switchToFrame(FRAME_DONT_EXISTS)).to.throw(Error);
+  });
+});
```

Comment:
there is another use case that we might want to test. Some iframes requires time to load. 
after switching to the iframe it's content not immediately loaded. we might want to test it as well

Reply:
How du suggest we do that?